optaplanner-core/src/test/java/org/optaplanner/core/impl/exhaustivesearch/ExhaustiveSearchTest.java

```diff
@@ -0,0 +1,363 @@
+package org.optaplanner.core.impl.exhaustivesearch;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import org.assertj.core.api.Assertions;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.optaplanner.core.api.solver.Solver;
+import org.optaplanner.core.api.solver.SolverFactory;
+import org.optaplanner.core.config.exhaustivesearch.ExhaustiveSearchPhaseConfig;
+import org.optaplanner.core.config.exhaustivesearch.ExhaustiveSearchType;
+import org.optaplanner.core.config.exhaustivesearch.NodeExplorationType;
+import org.optaplanner.core.config.heuristic.selector.common.SelectionCacheType;
+import org.optaplanner.core.config.heuristic.selector.common.SelectionOrder;
+import org.optaplanner.core.config.heuristic.selector.entity.EntitySelectorConfig;
+import org.optaplanner.core.config.heuristic.selector.entity.EntitySorterManner;
+import org.optaplanner.core.config.heuristic.selector.move.generic.ChangeMoveSelectorConfig;
+import org.optaplanner.core.config.heuristic.selector.value.ValueSelectorConfig;
+import org.optaplanner.core.config.heuristic.selector.value.ValueSorterManner;
+import org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig;
+import org.optaplanner.core.config.solver.SolverConfig;
+import org.optaplanner.core.config.solver.termination.TerminationConfig;
+import org.optaplanner.core.impl.exhaustivesearch.event.ExhaustiveSearchListener;
+import org.optaplanner.core.impl.solver.DefaultSolver;
+import org.optaplanner.core.impl.testdata.domain.comparable.TestdataComparableEntity;
+import org.optaplanner.core.impl.testdata.domain.comparable.TestdataComparableSolution;
+import org.optaplanner.core.impl.testdata.domain.comparable.TestdataComparableValue;
+import org.optaplanner.core.impl.testdata.score.director.TestdataComparableDifferentValuesCalculator;
+import org.optaplanner.core.impl.testdata.util.PlannerTestUtils;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(Parameterized.class)
+public class ExhaustiveSearchTest {
```

Comment:
This test deserves a Javadoc. Something explaining how the test was put together, what the magic values mean (`13--`, ...) and that they are, indeed, painstakingly manually crafter.